install/helm.go

```diff
@@ -125,12 +125,8 @@ func (k *K8sInstaller) generateManifests(ctx context.Context) error {
 			// TODO(gandro): Future versions of Cilium will remove the following
 			// two limitations, we will need to have set the config map values
 			// based on the installed Cilium version
-			helmMapOpts["l7Proxy"] = "false"
-			k.Log("ℹ️  L7 proxy disabled due to Wireguard encryption")
-
-			if k.params.NodeEncryption {
-				k.Log("⚠️️  Wireguard does not support node encryption yet")
-			}
+			helmMapOpts["l7Proxy"] = "true"
```

Comment:
I think we don't need to set any `l7Proxy` here, as it should be enabled by default.

However, we need to figure out which version of Cilium we are trying to install. I.e., we need to keep the existing code for <1.14 versions.

Reply:
@brb Thanks for the review. That means I just need to implement a simple `versioncheck ("<=1.13.0")` to restrict, right? 👇🏼 

```go
switch k.params.Encryption {
case encryptionIPsec:
	helmMapOpts["encryption.enabled"] = "true"
	helmMapOpts["encryption.type"] = "ipsec"
	if k.params.NodeEncryption {
		helmMapOpts["encryption.nodeEncryption"] = "true"
	}
case encryptionWireguard:
	helmMapOpts["encryption.enabled"] = "true"
	helmMapOpts["encryption.type"] = "wireguard"
	// TODO(gandro): Future versions of Cilium will remove the following
	// two limitations, we will need to have set the config map values
	// based on the installed Cilium version
	versioncheck.MustCompile("<=1.13.0")(k.chartVersion):
		helmMapOpts["l7Proxy"] = "false"
		k.Log("ℹ️  L7 proxy disabled due to Wireguard encryption")

		if k.params.NodeEncryption {
			k.Log("⚠️️  Wireguard does not support node encryption yet")
		}
}
```

Reply:
Yep :+1: